File: bubble_sort/bubble_sort_optimized.js
const bubble_sort_algorithm = (arr) => {
  console.log("Given array : ");
  console.log(arr);

  let size = arr.length - 1;
  let swapped = false;

  for (let i = 0; i < size; i++) {
    for (let j = 0; j < size - i; j++) {
      if (arr[j] > arr[j + 1]) {
        temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        swapped = true;
      }
    }
    if (!swapped) {
      break; // array is already sorted
    }
  }

  console.log("Sorted array : ");
  console.log(arr);
};

bubble_sort_algorithm([45, 3, -1, 63, 2, 3, 87, 1, 200]);

File: bubble_sort/bubble_sort_optimized.py
def bubble_sort(arr):

    size = len(arr) - 1
    swapped = False
    for i in range(size):
        for j in range(size-i):
            if arr[j] > arr[j+1]:
                arr[j],arr[j+1] = arr[j+1], arr[j]
                swapped = True
        if not swapped:
            break #array is already sorted


if __name__ == '__main__':
    elements = [5,9,2,1,67,34,88,34]
    print("Given array : " )
    print(elements)
    bubble_sort(elements)
    print("Sorted array : " )
    print(elements)